packages/template-retail-react-app/app/hooks/use-einstein.js

```
@@ -120,13 +122,14 @@ export class EinsteinAPI {
 
         let response
         try {
-            response = await fetch(`${this.host}/v3${endpoint}`, {
-                method: method,
-                headers: headers,
-                ...(body && {
-                    body: JSON.stringify(body)
+            if (this.dntPreference === false)
```

Let's make this `einsteinFetch` fail early like this:
```js
    async einsteinFetch(endpoint, method, body) {
        if (this.dntPreference !== false) return {}
        // ...
    }
```
Otherwise, the `response` can be left undefined and can error out in the lines below.